src/common/types/time.cpp

```diff
@@ -92,11 +92,19 @@ bool Time::TryConvertInternal(const char *buf, idx_t len, idx_t &pos, dtime_t &r
 		pos++;
 		// we expect some microseconds
 		int32_t mult = 100000;
+		if (nanos) {
+			// do we expect nanoseconds?
+			mult *= Interval::NANOS_PER_MICRO;
+		}
 		for (; pos < len && StringUtil::CharacterIsDigit(buf[pos]); pos++, mult /= 10) {
 			if (mult > 0) {
 				micros += (buf[pos] - '0') * mult;
```

Comment:
Should this not be a TryMultiply, or are we still guaranteed to never exceed an `int32_t` ?

Reply:
Good question but yes, we are guaranteed to have 9 or fewer digits, which is < 2^30. There are actually tests that would crash in debug if this was not true (.987654321)